service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/dataproccluster/startup.sh

```diff
@@ -0,0 +1,768 @@
+# Otherwise if an error occurred write the line and command that failed to guest attributes.
+function exit_handler {
+  local exit_code="${1}"
+  local line_no="${2}"
+  local command="${3}"
+  # Success! Set the guest attributes and exit cleanly
+  if [[ "${exit_code}" -eq 0 ]]; then
+    set_guest_attributes "${STATUS_ATTRIBUTE}" "COMPLETE"
+    exit 0
+  fi
+  # Write error status and message to guest attributes
+  set_guest_attributes "${STATUS_ATTRIBUTE}" "ERROR"
+  set_guest_attributes "${MESSAGE_ATTRIBUTE}" "Error on line ${line_no}, command \"${command}\". See ${POST_STARTUP_OUTPUT_FILE} for more information."
+  exit "${exit_code}"
+}
+trap 'exit_handler $? $LINENO $BASH_COMMAND' EXIT
+
+#######################################
+### Begin environment setup 
+#######################################
+
+# Let the UI know the script has started
+set_guest_attributes "${STATUS_ATTRIBUTE}" "STARTED"
+
+# Add dataproc user to sudoers
+emit "Adding dataproc user to sudoers"
+sudo usermod -aG sudo $DATAPROC_USER
+
+# Remove default user bashrc to ensure that the user's bashrc is sourced in non interactive shells
+${RUN_AS_DATAPROC_USER} "rm -f '${USER_BASHRC}'"
+${RUN_AS_DATAPROC_USER} "touch '${USER_BASHRC}'"
+
+emit "Resynchronizing apt package index..."
+
+# The apt package index may not be clean when we run; resynchronize
+apt-get update
+
+# Create the target directories for installing into the HOME directory
+${RUN_AS_DATAPROC_USER} "mkdir -p '${USER_BASH_COMPLETION_DIR}'"
+${RUN_AS_DATAPROC_USER} "mkdir -p '${USER_HOME_LOCAL_BIN}'"
+${RUN_AS_DATAPROC_USER} "mkdir -p '${USER_HOME_LOCAL_SHARE}'"
+
+# As described above, have the ~/.bash_profile source the ~/.bashrc
+cat << EOF >> "${USER_BASH_PROFILE}"
+
+### BEGIN: Terra-specific customizations ###
+if [[ -e ~/.bashrc ]]; then
+  source ~/.bashrc
+fi
+### END: Terra-specific customizations ###
+
+EOF
+
+# Indicate the start of Terra customizations of the ~/.bashrc
+cat << EOF >> "${USER_BASHRC}"
+
+### BEGIN: Terra-specific customizations ###
+
+# Set the correct java installation to use
+export JAVA_HOME="${USER_HOME_LOCAL_DIR}"
+
+# Prepend "${USER_HOME_LOCAL_BIN}" (if not already in the path)
+if [[ ":\${PATH}:" != *":${USER_HOME_LOCAL_BIN}:"* ]]; then
+  export PATH="${USER_HOME_LOCAL_BIN}":"\${PATH}"
+fi
+EOF
+
+emit "Installing common packages via pip..."
+
+# Install common packages. Use pip instead of conda because conda is slow.
+${RUN_AS_DATAPROC_USER} "pip3 install --user \
+  dsub \
+  nbdime \
+  nbstripout \
+  pandas_gbq \
+  pre-commit \
+  pylint \
+  pytest"
+
+# Install nbstripout for the jupyter user in all git repositories.
+${RUN_AS_DATAPROC_USER} "nbstripout --install --global"
+
+# Install gcsfuse for bucket mounting
+# install gnupg and lsb-release (for gcsfuse)
+apt-get install -y \
+  gnupg \
+  lsb-release
+
+# install gcsfuse for bucket mounting
+export GCSFUSE_REPO=gcsfuse-$(lsb_release -c -s) \
```

Comment:
Can the script add some detection of the existence of gcsfuse prior to trying to install it. Would be really useful to have a reusable block that could safely run elsewhere.
Was this pulled from https://cloud.google.com/storage/docs/gcsfuse-install? If so, let's add a link to that in the comments.